backend/remote-state/pg/backend.go

```diff
@@ -0,0 +1,114 @@
+package pg
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/hashicorp/terraform/backend"
+	"github.com/hashicorp/terraform/helper/schema"
+	_ "github.com/lib/pq"
+)
+
+const (
+	locksTableName  = "locks"
+	locksIndexName  = "locks_by_name"
+	statesTableName = "states"
+	statesIndexName = "states_by_name"
+)
+
+// New creates a new backend for Postgres remote state.
+func New() backend.Backend {
+	s := &schema.Backend{
+		Schema: map[string]*schema.Schema{
+			"conn_str": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Postgres connection string; a `postgres://` URL",
+			},
+
+			"lock": &schema.Schema{
```

Comment:
Is there any reason to make locking optional here? S3 is optional only because it requires DynamoDB too, and the S3 remote state existed before locking.

Reply:
Nice, I wondered why it was optional, but couldn’t determine the reasoning. Yes, I’ll remove the optional locking.

Reply:
Removed optional locking